SW_Weather.c

```diff
@@ -64,116 +68,703 @@
+    climateAverages->sdC4[1] = standardDeviation(climateOutput->frostFree_days, numYears);
+    climateAverages->sdC4[2] = standardDeviation(climateOutput->ddAbove65F_degday, numYears);
+    
+    // Calculate and set the standard deviation of cheatgrass variables
+    climateAverages->sdCheatgrass[0] = standardDeviation(climateOutput->PPT7thMon_mm, numYears);
+    climateAverages->sdCheatgrass[1] = standardDeviation(climateOutput->meanTempDriestQtr_C, numYears);
+    climateAverages->sdCheatgrass[2] = standardDeviation(climateOutput->minTemp2ndMon_C, numYears);
+}
 
-	wn->ppt[Yesterday] = wn->ppt[Today];
-	wn->rain[Yesterday] = wn->rain[Today];
+/**
+ @brief Calculate monthly and annual time series of climate variables from daily weather
+
+ This function has two different types of years:
+
+ - The first one being "calendar" years. Which refers to the days and months that show up on a calendar (Jan-Dec).
+
+ - The second type being "adjusted" years. This type is used when the site location is in the southern hemisphere
+ and the year starts in July instead of January. For example, the adjusted year 1980 starts half a year earlier on
+ July 1st 1979, and ends on June 30th, 1980 (see example below). If our data starts in 1980, then we don't have
+ values for the first six months of adjusted year, 1980. Therefore, we start calculating the climate variables in the
+ following adjusted year, 1981. And both the first and last six months of data are not used.
+
+ Calendar year vs. adjusted year:
+ | Calendar year | Year North | Start N | End N | Year South | Start S | End S |
+ |:------------------:|:---------------:|:---------:|:--------:|:---------------:|:---------:|:---------:|
+ | 1980 | 1980 | 1980 Jan 1 | 1980 Dec 31 | 1980 | 1979 July 1 | 1980 June 30 |
+ | 1981 | 1981 | 1981 Jan 1 | 1981 Dec 31 | 1981 | 1980 July 1 | 1981 June 30 |
+ | 1982 | 1982 | 1982 Jan 1 | 1982 Dec 31 | 1982 | 1981 July 1 | 1982 June 30 |
+ |    ...    |   ...   |        ...         |           ...        |    ...   |       ...           |         ...           |
+ | 2020 | 2020 | 2020 Jan 1 | 2020 Dec 31 | 2020 | 2020 July 1 | 2021 June 30 |
+ 
+ @param[in] allHist Array containing all historical data of a site
+ @param[in] numYears Number of years represented by `allHist`
+ @param[in] startYear Calendar year corresponding to first year of `allHist`
+ @param[out] climateOutput Structure of type SW_CLIMATE_CLIM that holds averages and
+ standard deviations output by `averageClimateAcrossYears()`
+ */
+
+void calcSiteClimate(SW_WEATHER_HIST **allHist, int numYears, int startYear,
+                     SW_CLIMATE_YEARLY *climateOutput, Bool inNorthHem) {
+
+    int month, yearIndex, year, day, numDaysYear, currMonDay;
+    int numDaysMonth, adjustedDoy, adjustedYear = 0, secondMonth, seventhMonth,
+    adjustedStartYear, calendarYearDays;
+    
+    double currentTempMin, currentTempMean, totalAbove65, currentJulyMin, JulyPPT,

```

Comment:
can we please replace all local variable names that contain "July" if we actually mean "7thMonth"?